core/src/main/java/nl/nn/adapterframework/jwt/JwtSecurityHandler.java

```diff
@@ -31,29 +31,30 @@ public class JwtSecurityHandler implements ISecurityHandler {
 
 	private final @Getter Map<String, Object> claimsSet;
 	private final @Getter String roleClaim;
-	private final @Getter String principalNameClaim;
+	private final @Getter String principalNameClaim; //Defaults to JWTClaimNames#SUBJECT
 
 	public JwtSecurityHandler(Map<String, Object> claimsSet, String roleClaim, String principalNameClaim) {
 		this.claimsSet = claimsSet;
 		this.roleClaim = roleClaim;
 		this.principalNameClaim = principalNameClaim;
 	}
 
+	//JWTClaimNames#AUDIENCE claim may be a String or List. Others are either a String or Long (epoch date)
 	@Override
 	public boolean isUserInRole(String role, PipeLineSession session) {
-		Object claim = getClaimsSet().get(roleClaim);
-		if(claim instanceof String){
+		Object claim = claimsSet.get(roleClaim);
+		if(claim instanceof String) {
 			return role.equals(claim);
-		} else if(claim instanceof List){
-			List<String> claimList = (List) claim;
-			return claimList.stream().anyMatch(s -> role.equals(s));
+		} else if(claim instanceof List) {
+			List<String> claimList = (List<String>) claim;
```

Comment:
Kan niet ook een list van longs zijn?

Reply:
> Kan dit niet ook een list van longs zijn?

Vast wel, we moeten dit waarschijnlijk als `List<Object>` casten en voor ieder element van de list `String.valueOf(element)` doen, om de string compare the kunnen doen.